Jacobi/Sensei/C/solution/JacobiDataAdaptor.cxx

```diff
@@ -232,6 +244,83 @@ int JacobiDataAdaptor::GetArrayName(const std::string &meshName, int association
   return 0;
 }
 
+int JacobiDataAdaptor::GetMeshHasGhostNodes(const std::string &meshName, 
+  bool &hasGhostNodes, int &nLayers)
+{
+  if (meshName != "mesh")
+    {
+    hasGhostNodes = false;
+    nLayers = 0;
+    SENSEI_ERROR("No mesh named " << meshName)
+    return -1;
+    }
+
+  hasGhostNodes = true;
+  nLayers = 1;
+  return 0;
+}
+
+int JacobiDataAdaptor::AddGhostNodesArray(vtkDataObject* mesh, const std::string &meshName)
+{
+  if (meshName != "mesh")
+    {
+    SENSEI_ERROR("No mesh named " << meshName)
+    return -1;
+    }
+
+  vtkMultiBlockDataSet *mb = dynamic_cast<vtkMultiBlockDataSet*>(mesh);
+  if (!mb)
+    {
+    SENSEI_ERROR("Invalid mesh type " << mesh->GetClassName())
+    return -1;
+    }
+
+  vtkRectilinearGrid *block = dynamic_cast<vtkRectilinearGrid*>(mb->GetBlock(0));
+  if (!block)
+    return -1;
+
+  int nx = this->sim->bx+2;
+  int ny = this->sim->by+2;
+  vtkUnsignedCharArray *gn = vtkUnsignedCharArray::New();
+  gn->SetNumberOfTuples(nx*ny);
+  gn->SetName(GHOST_NODE_ARRAY_NAME().c_str());
+  unsigned char *gptr = (unsigned char *)gn->GetVoidPointer(0);
+  memset(gn->GetVoidPointer(0), 0, nx*nx*sizeof(unsigned char));
+  unsigned char ghost = 1;
+  // Left column
+  if(this->sim->rankx > 0)
```

Comment:
all the patches have full set of ghost nodes, so I think these tests on rank is not needed